File: main/11_data_types.R
## Basic data types
# Check the type
typeof(1) # double
typeof('1') # character
typeof(TRUE) # logical
is.numeric(1) # true
is.character('1') # true

# Set the type
as.numeric('1') # 1
as.character(1) # '1'
as.logical(0) # false
as.factor(1) # 1; levels: 1

## Data types in columns
# Using base and/or tidyverse functions
library(tidyverse)
df = read_tsv('00_R_data_set.tsv')
str(df) # get info one data types
df$Age = as.character(df$Age)

File: main/11_data_types.py
## Basic data types
# Check the type
type(1) # int
type(1.5) # float
type('one') # string
type(True) # boolean

# Set the type
int(1.5) # 1
float(1) # 1.5
str(1) # '1'
bool(0) # False

## Data types in columns
# pandas
import pandas as pd
df = pd.read_csv('00_py_data_set.tsv', sep='\t')
df.dtypes # shows the type for each column
df.info() # alternative (more informative)
df['Age'] = df['Age'].astype(str) # types as above
